### DISCUSSION TITLE
Multi-region support, or an alternative?

### DISCUSSION THREAD

**Discussion by RaymondMoay** [supabase/supabase#4236] [Questions] [answered]
I understand that multi-region support for Supabase is currently not available. Its a feature I would DEFINITELY pay for, but I understand that there are more pressing features to build. I'm quite new to backend dev hence I'm here seeking for advice!

How would global apps built on Supabase handle the latency issue? My DB is currently based in the SG server but a recent flood of high-quality US users (some paying) are complaining of slow API calls. I could alternatively migrate my database over to a US one, however this would definitely slow down Asia users, and not to mention my development speed.

Any advice would be great!

EDIT: I've been exploring and thinking about retaining my servers in SG but deploy my serverless functions in SG instead of in the States.

**Comment by LoopsGod** [comment]
Depending on the nature of your data, you could mitigate this issue by using a Redis cache / Memcached / KV Store. 

Create endpoints for your database with `Cloudflare Workers` using the `supabase-js` package, and cache every result with an expiry in Cloudflare KV: https://developers.cloudflare.com/workers/runtime-apis/kv#creating-expiring-keys

Cloudflare replicates the data worldwide in < 60 seconds, so your cached results will be available scalable worldwide after this period. These in-memory cache keys query in < 1 ms, so your API will become blazing fast worldwide!

Good luck

**Comment by awalias** [comment, marked as answer]
Supabase now has Read Replicas : https://supabase.com/docs/guides/platform/read-replicas
